### DISCUSSION TITLE
Why larger VRAM get OOM with same setting, while smaller VRAM work well

### DISCUSSION THREAD

**Discussion by ruiba168** [oobabooga/text-generation-webui#5030] [Q&A] [answered]
I am finetuning Llama2. It work on my laptop, which have 16G 3080 Nvidia GPU. But when i tested it on my desktop with 24G 3090 Nvida GPU, i got OOM. Sound extra 8G doesn't help. I didn't use 8bit/4bit quantization.  I copied whole folder in my laptop(16G GPU,no OOM ) to my desktop(24G GPU) , it still get OOM with same setting. Is there any possible reason for it?

**Comment by michusx** [comment, marked as answer]
You didn't provide details about used system so we can only guess...
One possible explanation might be that on laptop you have installed more recent Nvidia drivers. And latest releases of drivers have system memory swap for GPU activated by default to prevent OOM in user applications.
In Windows you can check Task Manager ->graphics processor section. After loading LLM if there is swaping enabled then you will see increased usage of shared graphics processor memory.

**Comment by ruiba168** [comment]
Thank michusx, You are right. It's related shared memory. They use both latest driver. My laptop has 16G VRAM and 64G RAM. So it can use 16+64/2=48G memory. My desktop has 24G VRAM and 32G RAM, it can use 24+32/2=40G. So my desktop has smaller memory to be used.
